### PR TITLE
Do not produce infinity values in synthetic source for range fields

### PR DESCRIPTION
**Pull request by lkts**
Currently if one of range bounds of `float_range` or `double_range` field has a default value, it is represented as `"-Infinity"` or `"Infinity"`. This is bad since infinities are not valid input values for these range fields and therefore reindexing such document will fail. One of the main goals of synthetic source is to be able to reindex documents so this needed to change.

This PR changes the implementation to always explicitly write `null` to synthetic source if range bound has a default value (see rest tests). Note that for consistency this is applied to all range types and not only to `float_range` and `double_range`.

As a nice bonus it allows us to not put "weird" values in source like `"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff"` and `"-292275055-05-16T16:47:04.192Z"`.

One downside here is that if customer explicitly provides a value that is equal to default range bound, synthetic source will still contain `null`. I don't think this is a common use case and there is nothing we can really do (except storing some additional metadata).

Contributes to #106460.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/mapper/DoubleRangeFieldMapperTests.java [hunk 1]
```diff
@@ -8,67 +8,14 @@
...
-                put("gte", 20.0);
-                put("lte", "Infinity");
-            }
-        });
-
-        var source = getSourceFor(mapping, inputValues);
-        var actual = source.source().get("field");
-        assertThat(actual, equalTo(expected));
+    public void testStuff() {
+        System.out.println(RangeType.DOUBLE.nextUp(Double.NEGATIVE_INFINITY));
```
Comment (kkrik-es) [inline server/src/test/java/org/elasticsearch/index/mapper/DoubleRangeFieldMapperTests.java:18]: Remove?

**Comment by kkrik-es** [comment]
Should this be treated as a breaking change? I believe it's desirable since we want docs that can be reindexed, but it should  go through the committee nonetheless?

File: server/src/test/java/org/elasticsearch/index/mapper/DoubleRangeFieldMapperTests.java (hunk 1 above)
Comment (lkts) [reply server/src/test/java/org/elasticsearch/index/mapper/DoubleRangeFieldMapperTests.java:18]: Oh thanks.

**Comment by lkts** [comment]
Synthetic source for ranges is not released yet so it shouldn't be breaking #107081.

**Review by kkrik-es — APPROVED** [review]

**Comment by lkts** [comment]
@elasticmachine update branch

**Review by martijnvg — APPROVED** [review]
LGTM

**Pull request merged**